learn/work/fairness-readmission/index.qmd

```diff
@@ -0,0 +1,517 @@
+
+The fairness metrics `demographic_parity()`, `equal_opportunity()`, and `equalized_odds()` all take values very close to zero for many models. Notably, the left-most models (i.e. the most performant models with respect to `roc_auc()`) happen to be among the most fair according to the metrics we've chosen.
+
+More concretely, we can rank the model configurations to examine only the most performant models:
+
+```{r}
+#| label: "rank-results"
+rank_results(wflow_set_fit, rank_metric = "roc_auc") %>%
+  filter(.metric == "roc_auc")
+```
+
+As in the plot above, we see that almost all of the most performant model configurations arise from the boosted tree modeling workflow. Let's examine the results for specifically the modeling workflow that encodes age as a number more thoroughly:
+
+```{r}
+#| label: "autoplot-age-bt"
+autoplot(wflow_set_fit, id = "age_bt")
+```
+
+The learning rate `learn_rate` seems to have a more pronounced effect on the results metrics than the number of randomly selected predictors `mtry`. As before, we see that the most performant models with respect to `roc_auc()` tend to be the most fair with respect to our fairness metrics. Further, the values of each of the fairness metrics plotted above seem highly correlated. 
+
+From the perspective of a practitioner hoping to satisfy various stakeholders, the fact that these metrics are highly correlated makes the model selection process much easier. We can choose one fairness metric that we'd like to optimize for, and likely end up with a near-optimal configuration for the other metrics as a byproduct.
+
+:::callout-note
+In machine learning fairness, "impossibility thereoms" show that fairness definitions "are not mathematically or morally compatible in general" [@mitchell2021]. More concretely, unless we live in a world with no inequality, there is no way to satisfy many definitions of fairness at once. However, recent research emphasizes that near-fairness among more limited sets of metrics, like the three we've used here, is both possible and relevant [@bell2023].
+:::
+
+To choose a model that performs well both with respect to a typical performance metric like `roc_auc()` and the fairness metrics we've chosen, we will:
+
+1) Find the "best" model with respect to `roc_auc()`.
+2) Choose the most fair model with respect to `equalized_odds()` among the models within one standard error of `roc_auc()` to the model we identified in step 1.
+
+We can do so by first aggregating performance metrics from the `"age_bt"` result with `collect_metrics()` and then using some tidyr and dplyr for data manipulation:
+
+<!-- 
+TODO: it'd be great if `select_by_one_std_err()` could "just work" here, i.e.  
+select_by_one_std_err(best_workflow, metric = "roc_auc", desc(equalized_odds))
+-->
+
+```{r}
+#| label: "find-best-params"
```

Comment:
I'm not sure that you want to introduce another concept here but this is a perfect use case for desirability functions.

Reply:
Absolutely. f6842b0. :)